lib/fb-user-filestore-client.js

```diff
@@ -56,65 +56,94 @@ class FBUserfilestoreClient extends FBJWTClient {
-   * Fetch user file
+   * Get url to download uploaded file from
    *
    * @param {string} userId
    * User ID
    *
-   * @param {string} userToken
+   * @param {string} fingerprint
+   * File fingerprint
+   *
+   * @return {string} url
+   */
+  getFetchUrl (userId, fingerprint) {
+    return this.createEndpointUrl(endpoints.fetch, {serviceSlug: this.serviceSlug, userId, fingerprint})
+  }
+
+  /**
+   * Fetch user file
+   *
+   * @param {object} args
+   * Fetch args
+   *
+   * @param {string} args.userId
+   * User ID
+   *
+   * @param {string} args.userToken
    * User token
    *
-   * @param {string} fingerprint
+   * @param {string} args.fingerprint
    * File fingerprint
    *
+   * @param {object} logger
+   * Bunyan logger instance
+   *
    * @return {promise<string>}
    * Promise resolving to file
    *
    **/
-  fetch (userId, userToken, fingerprint) {
-    const urlPattern = endpoints.fetch
+  async fetch (args, logger) {
+    const {userId, userToken, fingerprint} = args
+    const url = endpoints.fetch
     const serviceSlug = this.serviceSlug
 
-    /* eslint-disable camelcase */
-    const encrypted_user_id_and_token = this.encryptUserIdAndToken(userId, userToken)
-    /* eslint-enable camelcase */
+    const encrypted_user_id_and_token = this.encryptUserIdAndToken(userId, userToken) // eslint-disable-line camelcase
 
-    return this.sendGet(urlPattern, {serviceSlug, userId, fingerprint}, {encrypted_user_id_and_token}, true)
-      .then(json => {
-        const {file} = json
-        return Buffer.from(file, 'base64').toString()
-      })
+    const context = {serviceSlug, userId, fingerprint}
+    const payload = {encrypted_user_id_and_token}
+
+    const json = await this.sendGet({url, context, payload}, logger)
+    const {file} = json
+    return Buffer.from(file, 'base64').toString()
   }
 
   /**
    * Store user file
    *
-   * @param {string} userId
+   * @param {object} args
+   * Store args
+   *
+   * @param {string} args.userId
    * User ID
    *
-   * @param {string} userToken
+   * @param {string} args.userToken
    * User token
    *
-   * @param {string|buffer} file
+   * @param {string|buffer} args.file
    * User file
    *
-   * @param {object} policy
+   * @param {object} args.policy
    * Policy to apply to file
    *
-   * @param {number} [policy.max_size]
+   * @param {number} [args.policy.max_size]
    * Maximum file size in bytes
    *
-   * @param {string} [policy.expires]
+   * @param {string} [args.policy.expires]
    * Maximum file size in bytes
    *
-   * @param {array<string>} [policy.allowed_types]
+   * @param {array<string>} [args.policy.allowed_types]
    * Allowed mime-types
    *
+   * @param {object} logger
+   * Bunyan logger instance
+   *
    * @return {promise<undefined>}
    *
    **/
-  store (userId, userToken, file, policy) {
-    const urlPattern = endpoints.store
+  async store (args, logger) {
+    const {userId, userToken} = args
+    let {file, policy} = args
```

Comment:
Not sure the benefit of `policy = Object.assign({}, policy)` but if we can remove that, policy could be a const. Maybe it's for cloning purposes?

Reply:
Just ensuring that we don't change the policy that was passed in.